brut.apktool/apktool-lib/src/main/java/brut/androlib/Config.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  Copyright (C) 2010 Ryszard Wiśniewski <brut.alll@gmail.com>
+ *  Copyright (C) 2010 Connor Tumbleson <connor.tumbleson@gmail.com>
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package brut.androlib;
+
+import brut.androlib.exceptions.AndrolibException;
+import brut.util.OSDetection;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.logging.Logger;
+
+/**
+ * Option storage.
+ */
+public class Config {
+
+    private final static Logger LOGGER = Logger.getLogger(Config.class.getName());
+
+    // Build options
+    public boolean forceBuildAll = false;
+    public boolean forceDeleteFramework = false;
+    public boolean debugMode = false;
+    public boolean netSecConf = false;
+    public boolean verbose = false;
+    public boolean copyOriginalFiles = false;
+    public boolean updateFiles = false;
+    public boolean isFramework = false;
+    public boolean resourcesAreCompressed = false;
+    public boolean useAapt2 = false;
+    public boolean noCrunch = false;
+    public int forceApi = 0;
+    public Collection<String> doNotCompress;
+
+    // decode options
+
+    public final static short DECODE_SOURCES_NONE = 0x0000;
+    public final static short DECODE_SOURCES_SMALI = 0x0001;
+    public final static short DECODE_SOURCES_SMALI_ONLY_MAIN_CLASSES = 0x0010;
+
+    public short decodeSources = DECODE_SOURCES_SMALI;
+
+    public final static short DECODE_RESOURCES_NONE = 0x0100;
+    public final static short DECODE_RESOURCES_FULL = 0x0101;
+
+    public short decodeResources = DECODE_RESOURCES_FULL;
+    public final static short FORCE_DECODE_MANIFEST_NONE = 0x0000;
+    public final static short FORCE_DECODE_MANIFEST_FULL = 0x0001;
+
+    public short forceDecodeManifest = FORCE_DECODE_MANIFEST_NONE;
+
+    public final static short DECODE_ASSETS_NONE = 0x0000;
+    public final static short DECODE_ASSETS_FULL = 0x0001;
+
+    public short decodeAssets = DECODE_ASSETS_FULL;
+
+    public boolean analysisMode = false;
+
+    public int apiLevel = 0;
+
+    public boolean forceDelete = false;
+
+    public boolean keepBrokenResources = false;
+
+    public boolean baksmaliDebugMode = true;
+
+    // common options
+
+    /**
```

Comment:
I'm not sure why we docblock this one and next one, but the rest are grouped under decode/common/build